backend/src/main/java/com/hackner/musicompass/discogsapi/service/DiscogsArtistApiService.java

```diff
@@ -0,0 +1,40 @@
+package com.hackner.musicompass.discogsapi.service;
+
+import com.hackner.musicompass.discogsapi.model.DiscogsArtistSearchResults;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class DiscogsArtistApiService {
+
+    private RestTemplate restTemplate;
+
+    private String baseUrl = "https://api.discogs.com";
+
+    @Autowired
+    public DiscogsArtistApiService(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public DiscogsArtistSearchResults getDiscogsArtistByArtistName(String artistName, String accessToken) {
```

Comment:
you may directly want to include `private final DiscogsSecret discogsSecret` in this class if it is only needed here - it would make your service class easier to read

Reply:
I decided to move this to the helper class for the entity - see ur next comment.